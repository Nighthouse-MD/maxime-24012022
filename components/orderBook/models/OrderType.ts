enum OrderType {
    ask = "sell",
    bid = "buy"
}

export default OrderType;