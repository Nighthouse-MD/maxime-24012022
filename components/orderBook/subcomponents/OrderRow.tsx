import { memo } from 'react';
import { Row } from 'react-bootstrap';
import styles from '../OrderBook.module.scss';
import OrderType from '../models/OrderType';

import { useMediaQuery } from 'react-responsive'

const OrderRow = ({ volumePercentage, type, children }: OrderRowProps) => {
    const isSmallScreen = useMediaQuery({ query: '(max-width: 768px)' })
    const gradientDirection = isSmallScreen ? 'right' : type === OrderType.ask ? 'right' : 'left';
    const dynamicBackgroundStyle = `linear-gradient(to ${gradientDirection}, ${type === OrderType.ask ? styles.redBackground : styles.greenBackground} ${volumePercentage}%, transparent 0)`;

    return <Row className={styles.orderBookRow} style={{ background: dynamicBackgroundStyle }}>
        {children}
    </Row>;
};

interface OrderRowProps {
    volumePercentage: number;
    type: OrderType;
    children: React.ReactNode;
}

export default memo(OrderRow);