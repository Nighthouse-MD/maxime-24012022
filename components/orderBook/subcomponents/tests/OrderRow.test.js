import OrderRow from '../OrderRow';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import OrderType from '../../models/OrderType';

describe('OrderRow', () => {
    it('shows the correct order for OrderType.ask', () => {
        const { container } = render(<OrderRow type={OrderType.ask} volumePercentage={43} children={<div>TestChild</div>} />);
        const theRow = container.getElementsByClassName("orderBookRow")[0];

        expect(theRow.className).toBe("orderBookRow row");
        expect(theRow).toHaveStyle("background:linear-gradient(to right, redBackground 43%, transparent 0)");
    });

    it('renders correctly for OrderType.ask', () => {
        const tree = renderer
            .create(<OrderRow type={OrderType.ask} volumePercentage={43} children={<div>TestChild</div>} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('shows the correct order for OrderType.bid', () => {
        const { container } = render(<OrderRow type={OrderType.bid} volumePercentage={43} children={<div>TestChild</div>} />);
        const theRow = container.getElementsByClassName("orderBookRow")[0];

        expect(theRow.className).toBe("orderBookRow row");
        expect(theRow).toHaveStyle("background:linear-gradient(to left, greenBackground 43%, transparent 0)");
    });

    it('renders correctly for OrderType.bid', () => {
        const tree = renderer
            .create(<OrderRow type={OrderType.bid} volumePercentage={43} children={<div>TestChild</div>} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});