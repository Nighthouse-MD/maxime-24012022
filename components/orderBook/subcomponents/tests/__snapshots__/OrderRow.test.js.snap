// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderRow renders correctly for OrderType.ask 1`] = `
<div
  className="orderBookRow row"
  style={
    Object {
      "background": "linear-gradient(to right, redBackground 43%, transparent 0)",
    }
  }
>
  <div>
    TestChild
  </div>
</div>
`;

exports[`OrderRow renders correctly for OrderType.bid 1`] = `
<div
  className="orderBookRow row"
  style={
    Object {
      "background": "linear-gradient(to left, greenBackground 43%, transparent 0)",
    }
  }
>
  <div>
    TestChild
  </div>
</div>
`;
